docs/book/v3/form-element-manager.md

```diff
@@ -0,0 +1,318 @@
+
+```php
+$element = $formElementManager->get(
+    ExampleElement::class,
+    [
+        'name'    => '…',
+        'options' => [ 
+            // …
+        ],
+    ]
+);
+```
+
+## Register and Fetch a Custom Element with a Factory
+
+If a custom element needs some preparation for creation then a factory can be used.
+
+The following example creates a class as factory for the element:
+
+```php
+final class ExampleElementFactory
+{
+    public function __invoke(Psr\Container\ContainerInterface): ExampleElement
+    {
+        $element = new ExampleElement();
+        $element->setOption('example_param', 'value');
+        
+        return $element;
+    }
+}
+```
+
+Register the custom element on the form element manager by using a factory class:
+
+```php
+$formElementManager->setFactory(
+    ExampleElement::class,
+    ExampleElementFactory::class
+);
+```
+
+Get the custom element and the option:
+
+```php
+$element = $formElementManager->get(ExampleElement::class);
+
+echo $form->getOption('example_param'); // value
+```
+
+Now the custom factory will be used to instance the element.
+
+## Fetch a Form without Registration
+
+The form element manager handles also whole forms.
+
+The following example creates a class as form:
+
+```php
+final class ExampleForm extends Laminas\Form\Form
+{
+    public function init(): void
+    {
+        // …
+    }
+}
+```
+
+The form element manager can create the form by the related class name:
+
+```php
+$form = $formElementManager->get(ExampleForm::class);
+```
+
+If no separate factory is required, then the form element manager will be instantiating the form class directly, by using the standard factory for elements (`Laminas\Form\ElementFactory`).
+
+## Register and Fetch a Form With a Factory
+
+If a form needs some preparation for creation then a factory can be used.
+
+The following example creates a class as factory for the form:
+
+```php
+final class ExampleFormFactory
+{
+    public function __invoke(Psr\Container\ContainerInterface): ExampleForm
+    {
+        $form = new ExampleForm();
+        $form->setName('example');
+        
+        return $form;
+    }
+}
+```
+
+Register the form on the form element manager by using a factory class:
+
+```php
+$formElementManager->setFactory(
+    ExampleForm::class,
+    ExampleFormFactory::class
+);
+```
+
+Get the form and the name:
+
+```php
+$form = $formElementManager->get(ExampleForm::class);
+
+echo $form->getName(); // example
+```
+
+Now the custom factory will be used to instance the form.
+
+## Set a Paramater to a Form on Instantiation
+
+The options of a form can be used to set custom parameters to a form.
+
+* It can be hooked into the initialization via the `init()` method.
+* The form element factory calls the `init()` method after creating an instance of the class.
+* Since the name and options are processed in the constructor, it can be accessed them via `init()` to perform further customizations for the instance.
```

Comment:
```suggestion
* It can be hooked into the initialization via the `init()` method.
* The form element factory calls the `init()` method after creating an instance of the class.
* Since the name and options are processed in the constructor, it can access them within `init()` to perform further customization of the instance.
```